library/HTMLPurifier/Filter/URIFilter/SafeLink.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+class HTMLPurifier_URIFilter_SafeLink extends HTMLPurifier_URIFilter
+{
+    /**
+     * @type string
+     */
+    public $name = 'SafeLink';
+
+    /**
+     * @type bool
+     */
+    public $always_load = true;
+
+    /**
+     * @type string
+     */
+    protected $regexp = null;
+
+    /**
+     * @param HTMLPurifier_Config $config
+     * @return bool
+     */
+    public function prepare($config)
+    {
+        $this->regexp = $config->get('URI.SafeLinkRegexp');
+
+        return true;
+    }
+
+    /**
+     * @param HTMLPurifier_URI $uri
+     * @param HTMLPurifier_Config $config
+     * @param HTMLPurifier_Context $context
+     * @return bool
+     */
+    public function filter(&$uri, $config, $context)
+    {
```

Comment:
Filtering should be controlled by `HTML.Link` config flag, similarly to `HTMLPurifier_URIFilter_SafeIframe`.

Reply:
After giving it some thought, I believe `HTML.SafeLink` is a better name (and is consistent with `HTML.Safe*` convention used by HTMLPurifier).